### PR TITLE
feat(cli): add strict prompt to model generator

### PR DESCRIPTION
**Pull request by nabdelgadir**
Connect to https://github.com/strongloop/loopback-next/issues/1530.

Add a prompt to model generator that asks the user if they want to allow additional properties (disable the `strict` setting). And add `modelSettings` to `this.artifactInfo` in `ModelGenerator` that contains settings such as `strict`.

## Checklist

- [x] `npm test` passes on your machine
- [x] New tests added or existing tests modified to cover all changes
- [x] Code conforms with the [style guide](http://loopback.io/doc/en/contrib/style-guide.html)
- [ ] API Documentation in code was updated
- [x] Documentation in [/docs/site](../tree/master/docs/site) was updated
- [x] Affected artifact templates in `packages/cli` were updated
- [ ] Affected example projects in `examples/*` were updated

### CODE REVIEW

File: packages/cli/generators/model/templates/model.ts.ejs [hunk 1]
```diff
@@ -5,7 +5,11 @@ import {model, property} from '@loopback/repository';
 import {<%= modelBaseClass %>} from '.';
 <% } -%>
 
+<% if(disabledStrictMode) { -%>
```
Comment (jannyHou) [inline packages/cli/generators/model/templates/model.ts.ejs]: I think a more general way to get the model settings from the CLI is passing in a settings object, like

```ts
 <% if(settings) { -%>
 @model(<%= settings %>)
 <% } else { -%>
 @model()
 <% } -%>
```

just a suggestion :) can do it in a separate PR.

File: packages/cli/generators/model/index.js [hunk 2]
```diff
@@ -154,20 +154,43 @@ module.exports = class ModelGenerator extends ArtifactGenerator {
...
+        type: 'confirm',
+        default: false,
+        when: !this.artifactInfo.disabledStrictMode,
+      },
+    ])
+      .then(setting => {
+        Object.assign(this.artifactInfo, setting);
+
         this.log(
           `Let's add a property to ${chalk.yellow(
```
Comment (jannyHou) [inline packages/cli/generators/model/index.js]: Any reason why we log this message here? I feel it should be in the `promptPropertyName` function. While I also noticed that this code was in the `promptModelBaseClass` function, so I am a little bit confused...

File: packages/cli/generators/model/index.js [hunk 3]
```diff
@@ -154,20 +154,43 @@ module.exports = class ModelGenerator extends ArtifactGenerator {
...
+    return this.prompt([
+      {
+        name: 'disabledStrictMode',
+        message: 'Disable strict mode for this model?',
+        type: 'confirm',
+        default: false,
+        when: !this.artifactInfo.disabledStrictMode,
+      },
+    ])
+      .then(setting => {
```
Comment (jannyHou) [inline packages/cli/generators/model/index.js]: Correct me if wrong:
I assume the setting should be an object with the `disabledStrictMode` property like:
`{disabledStrictMode: true}`

I prefer to add a property called `modelSettings` in `this.artifactInfo` in case we support more configurable setting attributes in the future, like:

```ts
artifactInfo: {
  modelSettings: {
    disabledStrictMode: true
  }
}
```

And btw, why do we return the `setting` at the end of the prompt 🤔

File: packages/cli/generators/model/index.js (hunk 2 above)
Comment (nabdelgadir) [reply packages/cli/generators/model/index.js]: It's so that it appears in this order: 
```
lb4 model
? Model class name: model
? Please select the model base class Entity (A persisted model with an ID)
? Disable strict mode for this model? No
Let's add a property to Model
```
instead of:
```
lb4 model
? Model class name: model
? Please select the model base class Entity (A persisted model with an ID)
Let's add a property to Model
? Disable strict mode for this model? No
```

File: packages/cli/generators/model/index.js (hunk 3 above)
Comment (nabdelgadir) [reply packages/cli/generators/model/index.js]: Thanks, I'll update it accordingly and remove the return!

File: packages/cli/generators/model/templates/model.ts.ejs [hunk 4]
```diff
@@ -5,7 +5,11 @@ import {model, property} from '@loopback/repository';
...
+@model({settings: {strict: false}})
+<% } else { -%>
 @model()
+<% } -%>
 export class <%= className %> extends <%= modelBaseClass %> {
```
Comment (bajtos) [inline packages/cli/generators/model/templates/model.ts.ejs]: Please note it's not enough to disable strict mode in model setting, the TypeScript class must be changed to allow additional properties too.

Cross-posting the relevant part from https://github.com/strongloop/loopback-next/issues/1530#issuecomment-434203916

```ts
  // Add an indexer property to allow additional data
  [prop: string]: any;
```

File: packages/cli/generators/model/templates/model.ts.ejs [hunk 5]
```diff
@@ -18,6 +22,15 @@ export class <%= className %> extends <%= modelBaseClass %> {
   <%= key %><%if (!val.required) {%>?<% } %>: <%= val.tsType %>;
 
 <% }) -%>
+
```
Comment (bajtos) [inline packages/cli/generators/model/templates/model.ts.ejs]: I think this will create an empty line in the generated output, even when `allowAdditionalProperties` was set to `false`. Please move this empty line inside `<% if(allowAdditionalProperties) { -%>` branch.

File: packages/cli/generators/model/templates/model.ts.ejs [hunk 6]
```diff
@@ -18,6 +22,15 @@ export class <%= className %> extends <%= modelBaseClass %> {
...
+<% if(allowAdditionalProperties) { -%>
+  // Define well-known properties here
+
+
```
Comment (bajtos) [inline packages/cli/generators/model/templates/model.ts.ejs]: Do we need two empty lines here? I think prettier will remove the second line anyways, won't it? Can we keep a single empty line in this template?
